internal/controllers/watch/watch.go

```diff
@@ -27,6 +27,16 @@ func NewController(mgr ctrl.Manager) error {
 			client:         mgr.GetClient(),
 			refControllers: map[apiv1.ResourceRef]*KindWatchController{},
 		})
+	if err != nil {
+		return err
+	}
+
+	return ctrl.NewControllerManagedBy(mgr).
```

Comment:
Is it okay to just return the pruning controller here/

Reply:
`Complete()` returns an error, which is the only return value here. Any concerns (I might be unclear on the question)

Reply:
Ah, I though the controller itself was being returned, lgtm.